MCsquare/matRad_sparseBeamletsReaderMCsquare.cpp

```diff
@@ -156,43 +155,47 @@ void mexFunction(   int nlhs,   mxArray *plhs[],    int nrhs,   const mxArray *p
             
             file.read((char *) data, numCurrentReadVals * sizeof(float));
             
-            for(uint32_t readIx = 0; readIx < numCurrentReadVals; ++readIx){
-                
+            for (uint32_t readIx = 0; readIx < numCurrentReadVals; ++readIx) {
+
                 //If necessary reallocate space for non-zeros
-                if(currentNnz >= nnzMaxEstimate){
+                if (currentNnz >= nnzMaxEstimate) {
                     oldNnzMaxEstimate = nnzMaxEstimate;
                     percent_sparse += 0.001;
-                    nnzMaxEstimate = (mwSize) std::ceil((double)numVoxels * (double)numSpots * percent_sparse);
-                    
-                    if (nnzMaxEstimate <= oldNnzMaxEstimate) nnzMaxEstimate = oldNnzMaxEstimate + 1;
-                    
+                    nnzMaxEstimate = (mwSize)std::ceil((double)numVoxels * (double)numSpots * percent_sparse);
+
+                    if (nnzMaxEstimate <= oldNnzMaxEstimate) 
+                        nnzMaxEstimate = oldNnzMaxEstimate + 1;
+
                     mxSetNzmax(plhs[0], nnzMaxEstimate);
-                    mxSetPr(plhs[0], (double *) mxRealloc(vals, nnzMaxEstimate*sizeof(double)));
-                    mxSetIr(plhs[0], (mwIndex *) mxRealloc(ix, nnzMaxEstimate*sizeof(mwIndex)));
-                    
-                    vals  = mxGetPr(plhs[0]);
+                    mxSetPr(plhs[0], (double *)mxRealloc(vals, nnzMaxEstimate * sizeof(double)));
+                    mxSetIr(plhs[0], (mwIndex *)mxRealloc(ix, nnzMaxEstimate * sizeof(mwIndex)));
+
+                    vals = mxGetPr(plhs[0]);
                     ix = mxGetIr(plhs[0]);
                 }
-                
+
                 // get index
                 ixMC2 = ixFirst + readIx;
-                    
-                // get subscripts
-                kSub = ixMC2/numVoxelsSlice;
-                jSub = (ixMC2 - kSub*numVoxelsSlice)/sizeDoseGrid[1];
-                iSub = ixMC2 - jSub*sizeDoseGrid[1] - kSub*numVoxelsSlice;
-                    
-                // flip image
-                iSub = sizeDoseGrid[0] - iSub - 1;
-
-                // compute new index
-                ixMatRad = jSub + iSub*sizeDoseGrid[0] + kSub*numVoxelsSlice;
-                
+
+                // get subscripts (NOTE: sizeDoseGrid is already permuted from the call within matRad)
+                kSub = ixMC2 / numVoxelsSlice;
+                jSub = (ixMC2 - kSub * numVoxelsSlice) / sizeDoseGrid[1];
+                //iSub = ixMC2 - jSub*sizeDoseGrid[1] - kSub*numVoxelsSlice;
```

Comment:
Why would you want to leave this in as comment if it is wrong?

Reply:
while I sometimes like to keep the wrong code in comments to see that there were issues at this point before.. but that is not necessary here I think. 
I can clean it up and update the fix.